cli/status/status.go

```diff
@@ -1,57 +1,185 @@
 	"github.com/tarantool/tt/cli/running"
 )
 
+//go:embed lua/instance_state.lua
+var replicationInfoLuaScript string
+
 // StatusOpts contains options for tt status.
 type StatusOpts struct {
 	// Option for pretty-formatted table output.
-	Pretty bool
+	Pretty  bool
+	Details bool
+}
+
+type Alert struct {
+	Type    string `mapstructure:"type"`
+	Message string `mapstructure:"message"`
+}
+
+type ConfigInfo struct {
+	Status string  `mapstructure:"status"`
+	Alerts []Alert `mapstructure:"alerts"`
+}
+
+type Upstream struct {
+	Status  string `mapstructure:"status"`
+	Message string `mapstructure:"message"`
+}
+
+type ReplicationInfo struct {
+	Name     string    `mapstructure:"name"`
+	Upstream *Upstream `mapstructure:"upstream"`
+}
+
+type InstanceState struct {
+	ReplicationInfo []ReplicationInfo `mapstructure:"replication_info"`
+	ConfigInfo      ConfigInfo        `mapstructure:"config_info"`
+	ReadOnly        string            `mapstructure:"read_only"`
+	BoxStatus       string            `mapstructure:"box_status"`
 }
 
 // Status writes the status as a table.
 func Status(runningCtx running.RunningCtx, opts StatusOpts) error {
 	ts := table.NewWriter()
 	ts.SetOutputMirror(os.Stdout)
-	ts.AppendHeader(table.Row{"INSTANCE", "STATUS", "PID", "MODE"})
+	ts.AppendHeader(
+		table.Row{"INSTANCE", "STATUS", "PID", "MODE", "CONFIG", "BOX", "UPSTREAM"})
+
+	instances := map[string]map[string]interface{}{}
+	instancesAlerts := map[string][]string{}
+
+	PrintYellow := color.New(color.FgYellow).SprintFunc()
+	PrintRed := color.New(color.FgRed).SprintFunc()
 
 	for _, run := range runningCtx.Instances {
-		row := []interface{}{}
 		fullInstanceName := running.GetAppInstanceName(run)
 		procStatus := running.Status(&run)
 
-		row = append(row, fullInstanceName)
-		row = append(row, procStatus.ColorSprint(procStatus.Status))
+		if _, exists := instances[fullInstanceName]; !exists {
+			instances[fullInstanceName] = map[string]interface{}{
+				"STATUS": "",
+				"PID":    nil,
+				"MODE":   "",
+				"CFG":    "--",
+				"BOX":    "--",
+				"UPS":    "--",
+			}
+		}
+
+		instances[fullInstanceName]["STATUS"] = procStatus.ColorSprint(procStatus.Status)
 		if procStatus.Code == process_utils.ProcessRunningCode {
-			row = append(row, procStatus.PID)
+			instances[fullInstanceName]["PID"] = procStatus.PID
 		}
 
 		conn, err := connector.Connect(connector.ConnectOpts{
 			Network: "unix",
 			Address: run.ConsoleSocket,
 		})
+
 		if err == nil {
-			res, err := conn.Eval("return (type(box.cfg) == 'function') or box.info.ro",
-				[]any{}, connector.RequestOpts{})
+			var instanceState InstanceState
+			isParsed := false
+			res, err := conn.Eval(replicationInfoLuaScript, []any{},
+				connector.RequestOpts{})
 			if err == nil && len(res) != 0 {
-				mode := res[0].(bool)
-				mode_str := "RO"
-				if !mode {
-					mode_str = "RW"
+				isParsed = mapstructure.Decode(res[0], &instanceState) == nil
+			}
+			if !isParsed {
+				continue
+			}
```

Comment:
Just thought: if we can't parse tarantool's state information, doesn't it mean that something likely is wrong with the instance and we should add something into the `--details` output?

Reply:
made errors more informative and covered more cases.